include/materials/InelasticRebar.h

```diff
@@ -0,0 +1,63 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "LinearElasticTruss.h"
+
+class InelasticRebar;
+
+template <>
+InputParameters validParams<InelasticRebar>();
+
+class InelasticRebar : public LinearElasticTruss
+{
+public:
+  InelasticRebar(const InputParameters & parameters);
+
+protected:
+  virtual void computeQpStrain();
+  virtual void computeQpStress();
+  virtual void initQpStatefulProperties();
+
+  virtual Real computeHardeningValue(Real scalar);
+  virtual Real computeHardeningDerivative(Real scalar);
+
+  const VariableValue & _poissons_ratio;
+  Real _yield_stress;
+
+  const Real _hardening_constant;
+  const Function * _hardening_function;
+  Real _hardening_slope;
+
+  /// Convergence tolerance
+  Real _absolute_tolerance;
+  Real _relative_tolerance;
+
+  const MaterialProperty<Real> & _total_stretch_old;
+  MaterialProperty<Real> & _plastic_strain;
+  const MaterialProperty<Real> & _plastic_strain_old;
+  MaterialProperty<Real> & _eqv_plastic_strain;
+  const MaterialProperty<Real> & _eqv_plastic_strain_old;
+  const MaterialProperty<Real> & _elastic_strain_old;
+  const MaterialProperty<Real> & _stress_old;
+  MaterialProperty<Real> & _strain_increment;
+
+  MaterialProperty<Real> & _hardening_variable;
+  const MaterialProperty<Real> & _hardening_variable_old;
+
+  /// maximum no. of iterations and iteration number
+  const unsigned int _max_its;
+  unsigned int _iteration;
```

Comment:
_iteration shouldn't be a member variable.  Just decleare it as a local variable in the function where it is used.